scripts/zones/Balgas_Dais/mobs/Wyrm.lua

```diff
@@ -4,5 +4,67 @@
 -- KSNM: Early Bird Catches the Wyrm
 -----------------------------------
 
+function onMobInitialize(mob)
+    mob:setMobMod(tpz.mobMod.DRAW_IN, 1) -- has a bug during flight, like Tiamat
+end
+
+function onMobSpawn(mob)
+    mob:setTP(3000) -- opens fight with a skill
+end
+
+function onMobFight(mob, target)
+    local isBusy = false
+    local action = mob:getCurrentAction()
+    local hasLanded = mob:getLocalVar("hasLanded")
+
+    if action == tpz.act.MOBABILITY_START or action == tpz.act.MOBABILITY_USING or action == tpz.act.MOBABILITY_FINISH then
+        isBusy = true -- when the Wyrm is in any stage of using a mobskill
+    end
+
+    -- Return to ground at 33% HP
+    if
+        mob:getHP() < mob:getMaxHP() * 0.33 and
+        mob:actionQueueEmpty() == true and
+        isBusy == false
+    then
+        -- After using Touchdown return to normal
+        if hasLanded == 1 then
+            -- Gets bonuses to Evasion, Defense and M.Evasion
+            mob:addStatusEffect(tpz.effect.EVASION_BOOST, 75, 0, 0)
+            mob:getStatusEffect(tpz.effect.EVASION_BOOST):setFlag(tpz.effectFlag.DEATH)
+            mob:addStatusEffect(tpz.effect.DEFENSE_BOOST, 75, 0, 0)
+            mob:getStatusEffect(tpz.effect.DEFENSE_BOOST):setFlag(tpz.effectFlag.DEATH)
+            mob:addStatusEffect(tpz.effect.MAGIC_DEF_BOOST, 75, 0, 0)
+            mob:getStatusEffect(tpz.effect.MAGIC_DEF_BOOST):setFlag(tpz.effectFlag.DEATH)
+            -- Melee attacks can hit again
+            mob:delStatusEffectSilent(tpz.effect.TOO_HIGH)
+            -- Reset default attack
+            mob:SetMobSkillAttack(0)
+            mob:SetMobAbilityEnabled(true)
+            mob:SetAutoAttackEnabled(true)
+            mob:AnimationSub(0)
+        -- Use Touchdown
+        elseif mob:AnimationSub() == 1 then
+            mob:SetMobAbilityEnabled(false) -- prevent overlaps
+            mob:SetAutoAttackEnabled(false)
+            mob:useMobAbility(1282)
+            mob:setLocalVar("hasLanded", 1)
+        end
+    
+    -- Go airborne at 66% HP
+    -- TODO: Should move physically to center/origin before taking off
+    elseif
+        mob:getHP() < mob:getMaxHP() * 0.66 and
+        mob:getHP() > mob:getMaxHP() * 0.33 and
+        mob:AnimationSub() == 0 and -- is on ground
+        mob:actionQueueEmpty() == true and
+        isBusy == false
+    then
+        mob:addStatusEffectEx(tpz.effect.TOO_HIGH, 0, 1, 0, 0) -- melee attacks miss
+        mob:AnimationSub(1) -- lift off
+        mob:SetMobSkillAttack(730) -- change default attack to ranged fire magic damage
```

Comment:
This will set it to use Tiamat's alternative skill list, which contains only Inferno Blast - not the Flame Blast you're wanting Wyrm to use.

Others: At some point we should rename this function to include the word "list".

Reply:
Every wyrm has its own set of skill IDs and there is NO OVERLAP. SE duplicated the IDs each and every time, even if the skill _names_ are identical. So even with the same moveset as Fafnir or Nidhogg, you do not use those IDs. you use the ones specifically for this Wyrm. Which is why faf and nid both have the ones associated with flying even though they do not - duplication!

Reply:
@ibm2431 I propose `mob:meleeSkillList()` Normal melee being 0. Its never unset or even `get`'d so no real point is specifying `set` in its name anyway.

Reply:
Ok first things first. That is not at all supposed to be Flame Blast. Flame Blast is included in the Wyrm's skill list with its own proper ID, had a script created and can be used as a normal mobskill during flight.
Tiamat and Wyrm are the only wyrms in Vana'diel that during flight use a fire spit-like magic based regular attack. There's also a wind and an ice version of this. Tiamat uses `SetMobSkillAttack` so I re-used this.
Indeed it brings some issues with it though. I observed and debugged that this actually makes the `mob` use a TP mobskill unconditionally now, using all TP on every regular hit. This is why I was having trouble getting Flame Blast to trigger during flight. The Wyrm would reset TP on every hit.
I will look into `meleeSkillList()`.

Reply:
much old stuff is not 100% accurate if that helps make sense of weirdness you see. There should (in the client) be a duplicate for each of them, because the way SE does it on their retail server and the client its associated with the model and thanks to ps2 limitations when they were in the design phase anytime they made a new model it spawned a new copy of everything along with it. So they clone Wyrm and color change it, poof we have Fafnir and they just disable the flight one their aren't using. They do it again and change the colors and strings for the skill names, hello Tiamat and Jormy. 

@Hozu & @MarianArlt 
Kjlotus used an unused blank skill to make Nid's "super buff" because Nid will never use that entry in the list normally, and the other wyrms won't use super buff. Any other Wyrm that does use it shouldn't be teh same list entries Nid uses (again, duplicates)

Reply:
elaborating a bit:
@MarianArlt basically, in early DSP we did not have teh ability to create additional skill lists, and we tied everything to a mob family ID. tahts since been corrected so we can do mob skill lists per mob just like retail, but not everything form back then is split up or even needs to be (we do not yet have a reason to create multiple separate screwdiver version for different puguils for instance, since they all work as is and players don't know the difference - good enough right?) But somewhere in the master list of skills, there are multiples for many things. I happen to know Wyrms moves are among them having made a bunch of custom ones for another server once upon a time.

Reply:
Can I mark this as resolved or do you want me to create a unique spell with this and then append a skill list entry for `KS99_Wyrm-Flying-Attack` in `mob_skill_lists.sql`?

Reply:
There aren't skill or magic like regular attacks (category 2). Any time you see a mob appear to use something like that for a "regular attack", they're mobskills (category 11) with a message of 0. They _look_ like a regular attack to a player, but internally they're different.

In this case, setting a skill list ID of 730 will cause it to use Tiamat's Inferno Blast "alt".
![skill](https://user-images.githubusercontent.com/13112942/96431219-bb799f80-11f2-11eb-8500-cc2adebb67fb.png)

Reply:
I know that ibm, that's exactly what SetMobSkillAttack() seem to be created for. Quite some wyrms use it. Where are we going with this?

Reply:
With Wyrms SE put minor variations while they were copy pasting, I can see how my use of the word duplicate has caused some confusion. Tiamats skills are no more KS99_Wyrms skills than they are Jormies, but SE _duplicated_ their template Wyrm to create each of them, if hat makes sense, and lead to certain Wyrms having unused entries - like Fafnir isn't going to touchdown cause he never flies, and that one in abyssea never will coz he or she never lands and this is what I was trying to say, Their breaths and blasts are probably not going to be 100% identical even if the same element.

Reply:
whatever ID retail is using, is the ID we want to use.

Reply:
I have an issue with this. As you can see in in the latest commits I have corrected all the Wyrm specific IDs with their corresponding animation IDs and created two additional skill lists. One for TP skills while flying, and one for the regular attack while flying. These three animations can be called individually fine and work as expected when called with `useMobAbility()` for example. **However**: When I set `mob:SetMobSkillAttack(1146)` I would expect the Wyrm to now use skillID 950 as regular attack, which has the correct animationID and the same values as any other of the flying-attack skills (leet, ocher etc.). **But** the mob just won't use anything at all. It just sits there doing nothing now.
I made sure that the database was correct, including the changes, the right IDs etc.

Reply:
> does absolutely nothing despite having added new skill lists exclusively for this use, using the correct skill, the correct animation ID etc. database double checked.

its expecting a script with the name that is on skill id 950 that is in that skill list: `flame_blast_alt`

Reply:
I thought that too, but how is it possible that `inferno_blast_alt`, `sleet_blast_alt` and `ochre_blast_alt` all work **without** having a script?

Reply:
bold of you to think somebody didn't ~~break~~ "fix" jormy.

try it. copy you blast to the other name and see if it starts blasting.

Reply:
Finally resolved after a few days of overthinking and a little chat. Thank you.